zio-http/shared/src/main/scala/zio/http/template2/Dom.scala

```diff
@@ -0,0 +1,476 @@
+
+    def whenSome[T](option: Option[T])(f: T => Seq[Modifier]): Style =
+      option.fold(this)(value => apply(f(value): _*))
+
+    // Style-specific methods
+    def css(code: String): Style = apply(Dom.text(code))
+
+    def media(value: String): Style = attr("media", value)
+
+    def `type`(value: String): Style = attr("type", value)
+
+    def scoped: Style = attr("scoped", true)
+
+    def inlineCss(code: String): Style = `type`("text/css").css(code)
+
+    private[template2] def renderInternal(state: RenderState): CharSequence =
+      renderElementInternal(state, shouldEscapeContent)
+  }
+}
+  final case class Text(content: String) extends Dom {
+    private[template2] def renderInternal(state: RenderState): CharSequence =
+      htmlEscape(content)
+  }
+
+  final case class RawHtml(content: String) extends Dom {
+    private[template2] def renderInternal(state: RenderState): CharSequence = content
+  }
+
+  case object Empty extends Dom {
+    private[template2] def renderInternal(state: RenderState): CharSequence = ""
+  }
+
+  final case class Fragment(children: Iterable[Dom]) extends Dom {
+    def apply(children: Dom*): Fragment =
+      Fragment(this.children ++ children)
+
+    private[template2] def renderInternal(state: RenderState): CharSequence =
+      children.map(_.renderInternal(state)).mkString(state.separator)
+  }
+
+  // Attribute types - all extend Modifier
+  sealed trait Attribute extends Modifier {
+    def name: String
+    def value: AttributeValue
+  }
+
+  /**
+   * Complete attribute with name and value
+   */
+  final case class CompleteAttribute(name: String, value: AttributeValue) extends Attribute
+
+  /**
+   * Partial attribute that needs a value (like href := "...")
+   */
+  final case class PartialAttribute(name: String) {
+    def :=(value: AttributeValue): CompleteAttribute = CompleteAttribute(name, value)
+    def :=(value: String): CompleteAttribute = CompleteAttribute(name, AttributeValue.StringValue(value))
+    def :=(value: Boolean): CompleteAttribute = CompleteAttribute(name, AttributeValue.BooleanValue(value))
+    def :=(value: Int): CompleteAttribute = CompleteAttribute(name, AttributeValue.StringValue(value.toString))
+    def :=(value: Double): CompleteAttribute = CompleteAttribute(name, AttributeValue.StringValue(value.toString))
+  }
+
+  /**
+   * Boolean attribute (like disabled, checked, etc.)
+   */
+  final case class BooleanAttribute(name: String, enabled: Boolean = true) extends Attribute {
+    def value: AttributeValue = AttributeValue.BooleanValue(enabled)
+  }
+
+  // Attribute value types
+  sealed trait AttributeValue extends Product with Serializable
+  object AttributeValue {
+    final case class StringValue(value: String) extends AttributeValue {
+      override def toString: String = value
+    }
+    final case class BooleanValue(value: Boolean) extends AttributeValue {
+      override def toString: String = value.toString
```

Comment:
Avoids need for boxing
```suggestion
      override def toString: String = String.valueOf(value)
```